llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1844,23 +1854,30 @@ class VPWidenInductionRecipe : public VPHeaderPHIRecipe {
 class VPWidenIntOrFpInductionRecipe : public VPWidenInductionRecipe {
   TruncInst *Trunc;
 
+  bool isUnrolled() const { return getNumOperands() == 6; }
+
 public:
   VPWidenIntOrFpInductionRecipe(PHINode *IV, VPValue *Start, VPValue *Step,
-                                VPValue *VF, const InductionDescriptor &IndDesc,
-                                DebugLoc DL)
+                                VPValue *VF, VPValue *StepVector,
+                                const InductionDescriptor &IndDesc, DebugLoc DL)
       : VPWidenInductionRecipe(VPDef::VPWidenIntOrFpInductionSC, IV, Start,
                                Step, IndDesc, DL),
         Trunc(nullptr) {
     addOperand(VF);
+    // Dummy StepVector replaced in convertToConcreteRecipes
+    addOperand(StepVector);
   }
 
   VPWidenIntOrFpInductionRecipe(PHINode *IV, VPValue *Start, VPValue *Step,
-                                VPValue *VF, const InductionDescriptor &IndDesc,
+                                VPValue *VF, VPValue *StepVector,
+                                const InductionDescriptor &IndDesc,
                                 TruncInst *Trunc, DebugLoc DL)
       : VPWidenInductionRecipe(VPDef::VPWidenIntOrFpInductionSC, IV, Start,
                                Step, IndDesc, DL),
         Trunc(Trunc) {
     addOperand(VF);
+    // Dummy StepVector replaced in convertToConcreteRecipes
```

Comment:
Comment doesn't seem accurate now, as we pass in `StepVector`. Could we still create the poison value here and maybe say something like `Temporarily use Poison for step-vector, which will only be introduced when needed, when preparing to execute.`

Also, should we exclude it from printing, if poison?

Reply:
To create poison we need access to the VPlan though. Is there a good way to access it from the constructor here?

Reply:
Ah yes, don't think there's a good way to retrieve it here; how about introducing stepvector just before unrolling, similarly to `materializeBroadcasts`? Would that work without the need of adding a dummy operand up-front?

Reply:
Yes that seems to work, I've done that in 8201fa2fc246d2187acaafa6030e2abc33811e56